src/app/metadata/features/global-summary/global-summary.component.ts

```diff
@@ -47,5 +48,25 @@ export class GlobalSummaryComponent {
   datasets = computed(() => this.stats().Dataset);
   methods = computed(() => this.stats().ExperimentMethod);
   individuals = computed(() => this.stats().Individual);
-  files = computed(() => this.stats().ProcessDataFile);
+  files: Signal<FileStatsModel> = computed(() => {
+    let fileCount = 0;
+    const formats: { value: string; count: number }[] = [];
```

Comment:
It would be better to make `formats` a `Map` (from value to count). Then you can avoid the slow find operations, and use gets instead. At the end, you can convert the Map back to the array of value/count objects.